File: src/main/java/com/akagiyui/drive/model/response/PermissionResponse.java
package com.akagiyui.drive.model.response;

import com.akagiyui.drive.model.Permission;
import lombok.Data;
import lombok.experimental.Accessors;

import java.util.List;

/**
 * 权限响应
 *
 * @author AkagiYui
 */
@Data
@Accessors(chain = true)
public class PermissionResponse {
    /**
     * 权限名
     */
    private String name;

    /**
     * 权限描述
     */
    private String description;

    public PermissionResponse(Permission permission) {
        this.name = permission.name();
        this.description = permission.getDescription();
    }

    /**
     * 从权限列表转换
     *
     * @param permissions 权限列表
     * @return 权限响应列表
     */
    public static List<PermissionResponse> fromPermissionList(List<Permission> permissions) {
        return permissions.stream().map(PermissionResponse::new).toList();
    }

}

File: app/src/main/kotlin/com/akagiyui/drive/model/response/PermissionResponse.kt
package com.akagiyui.drive.model.response

import com.akagiyui.drive.model.Permission

/**
 * 权限响应
 *
 * @author AkagiYui
 */
class PermissionResponse(permission: Permission) {
    /**
     * 权限名
     */
    var name = permission.name

    /**
     * 权限描述
     */
    var description = permission.description

    companion object {
        /**
         * 从权限列表转换
         *
         * @param permissions 权限列表
         * @return 权限响应列表
         */
        fun fromPermissionList(permissions: List<Permission>): List<PermissionResponse> {
            return permissions
                .stream()
                .map { permission -> PermissionResponse(permission) }
                .toList()
        }
    }
}
